### ISSUE TITLE
Basic instructions to run/test the queueserver from the queue monitor would help too; e.g.

### ISSUE THREAD

**Issue by prjemian** [BCDA-APS/bs_model_instrument#3]
Basic instructions to run/test the queueserver from the queue monitor would help too; e.g. 

- Connect to the host
- Change tab, open the environment
- Open plan editor, add sim plans to queue
- Start queue
The simulated plan run so fast that  it is hard to tell if they actually ran, need to look at the console (on the other tab). Not trivial for someone who has never used the queueserver.

_Originally posted by @rodolakis in https://github.com/BCDA-APS/bs_model_instrument/issues/2#issuecomment-2417771698_

**Comment by prjemian** [comment]
The QS host interacts with `redis` at this network host:port: https://github.com/BCDA-APS/bs_model_instrument/blob/dda801d56acf454cc55b6bb96efdee4b4448eec4/qs/qs-config.yml#L13

QS clients do not need to know about this connection.  They interact only with the QS host.

**Comment by prjemian** [comment]
![image](https://github.com/user-attachments/assets/38b6f4db-4899-4963-88d9-9c15e31f7a38)

Right.  If the QS host wanted to use the redis service which was running on a different workstation (say `example.server.org`), then change `qs-config.yml` as you said:

```yml
  redis_addr: example.server.org:6379 
```

**Comment by prjemian** [comment]
> Basic instructions to run/test the queueserver ...

I'm sure we wrote these instructions in one of the BDP demo presentations.  I'll dig that up.

**Comment by rodolakis** [comment]
I am happy to write those


> > Basic instructions to run/test the queueserver ...
> 
> I'm sure we wrote these instructions in one of the BDP demo presentations. I'll dig that up.

**Comment by prjemian** [comment]
These instructions will need some translation into our current names.  They were used at the time of this presentation (2024-02-09).  For completeness, I'm copying the subset of slides.  Includes APS Data Management (which is an additional feature).


## general

![image](https://github.com/user-attachments/assets/2e5628a9-02ad-4a5e-bf56-65a2e5256799)

![image](https://github.com/user-attachments/assets/137f768d-2f79-426f-95d6-9141799b6b8d)

![image](https://github.com/user-attachments/assets/cd57d2d6-e6c3-4844-ad62-d10cfac5dbce)

To watch the QS Host console on your terminal, use this bash command: `qserver-console-monitor`.
Type ^C to exit this.

<details>

## when using data management

![image](https://github.com/user-attachments/assets/55d969e4-b17b-4145-8d30-ffd41ddb9dbe)

![image](https://github.com/user-attachments/assets/d86d9c5c-6560-48da-8759-f81706d572b3)

![image](https://github.com/user-attachments/assets/f43a76d5-cdb3-4d2b-9cd6-354c1bb81095)

![image](https://github.com/user-attachments/assets/0e0d34cc-63f4-4016-a09a-aa18b9b44c37)

![image](https://github.com/user-attachments/assets/eb3c06c2-0a23-4ccc-8b60-48e64a16cb1a)

![image](https://github.com/user-attachments/assets/1758d75d-2205-4c78-8d5c-d330676bcd82)

</details>

**Comment by prjemian** [comment]
QS Host and QS clients are separate processes.

- QS Host process is started from a workstation with access to:
  - the `qs-config.yml` file
  - a `redis` service
  - the instrument package with the RE, devices, and plans
- Any QS client is started from a workstation with network access to the QS Host process

**Comment by prjemian** [comment]
Your question involved the `qs-config.yml` file and port 6379 (port used by the redis service).    This configuration file is used when running `start-re-manager` (the program that actually operates the queueserver host process).

I'll look again at the `qs_host.sh` (which calls `start-re-manager`) to make sure that its comments and code agree with our documentation.  The shell script has some defense against certain beamline staff trying to start the QS host on random workstations.  (To avoid having multiple RE instances running at the same time, actively running plans on the same hardware.)  By team agreement, we chose one, specific workstation.  The code checks for that.  This additional configuration must be known and checked before `start-re-manager` is run.

**Comment by rodolakis** [comment]
> QS Host and QS clients are separate processes.
> 
> * QS Host process is started from a workstation with access to:
>   
>   * the `qs-config.yml` file
>   * a `redis` service
>   * the instrument package with the RE, devices, and plans

Right, I am perfectly good with that part. But I am still unclear about how, in practice, do you achieve that:

> * Any QS client is started from a workstation with network access to the QS Host process

Do you need  that (QS client) workstation to have the package installed with a different set of settings for `qs-config.yml`? Or a lighter version of the package (since it does not need to know about the RE, devices and plans, which are handled by the host)? How do you start the Queuemonitor from that workstation?

**Comment by prjemian** [comment]
QS clients will never use the `qs-config.yml` file.

**Comment by prjemian** [comment]
> Or a lighter version

The packages required depend on which client program is to be run.  For example, the standard GUI (`queue-monitor`) comes from the [`bluesky_widgets`](https://blueskyproject.io/bluesky-widgets/) package.

**Comment by prjemian** [comment]
To run a client, follow the installation steps for that client.  They don't need the same suite of packages that the server needs.

**Comment by prjemian** [comment]
Q: Does a QS client need to know about the redis server or the `qs-config.yml` file?
A: No.  It only needs to know how to communicate with the QS Host process.

**Comment by prjemian** [comment]
This is more a documentation issue and can be resolved after 1.0.0 release.

**Comment by prjemian** [comment]
Time to revisit this now, since one beamline asked today about how to connect a QS client on workstation B with a QS server on workstation A.

> I launched qserver / redis on a machine named `A` and I will like to connect to the qserver on a different machine on the same subnet, and it seems that I don't have zmq connection.
